Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/PerfLib.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Globalization;
+
+    internal class PerfLib : IQuickPulsePerfLib
+    {
+        private static PerfLib library = null;
+
+        private PerformanceMonitor performanceMonitor;
+
+        private PerfLib()
+        {
+        }
+
+        public static PerfLib GetPerfLib()
+        {
+            library = library ?? new PerfLib();
```

Comment:
Nit: For lazy initialization write this:
```C#
return library ?? (library = new PerfLib());
```
Note the inline assignment on the right hand side is not executed when the value is non-null (already initialized).
Although, frankly, you could just have ```private static readonly library = new PerfLib();``` and let the runtime deal with atomic, lazy initialization (see [beforefieldinit](https://www.bing.com/search?q=beforefieldinit))